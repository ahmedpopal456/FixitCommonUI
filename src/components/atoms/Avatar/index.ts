import { AvatarBadge } from './Avatar';
export { AvatarBadge};