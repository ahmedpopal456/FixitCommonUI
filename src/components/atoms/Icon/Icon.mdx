---
name: Icon
menu: Atoms
---

import { Props, Playground } from "docz";
import { Icon } from "./Icon.tsx";

# Icon

A Icon is a clickable view to carry out actions such as clearing user input or submitting the form.

# Usage

```tsx
import { Icon } from "fixit-mobile-components";

<Icon />;
```

## Props

<Props of={Icon}></Props>

## Default

<Playground>
  <Icon name="emoticon-kiss" />
</Playground>

## Size

The default size is 24.

<Playground>
  <Icon name="format-size" size={10} />
  <Icon name="format-size" size={20} />
  <Icon name="format-size" size={30} />
  <Icon name="format-size" size={40} />
  <Icon name="format-size" size={50} />
  <Icon name="format-size" size={60} />
  <Icon name="format-size" size={70} />
</Playground>

## Color

The colors must be from the Fixit color pallette.

<Playground>
  <Icon name="palette-swatch" color="primary" />
  <Icon name="palette-swatch" color="accent" />
  <Icon name="palette-swatch" color="grey" />
  <Icon name="palette-swatch" color="light" />
  <Icon name="palette-swatch" color="dark" />
  <Icon name="palette-swatch" color="red" />
  <Icon name="palette-swatch" color="orange" />
  <Icon name="palette-swatch" color="yellow" />
  <Icon name="palette-swatch" color="blue" />
  <Icon name="palette-swatch" color="green" />
  <Icon name="palette-swatch" color="notification" />
  <Icon name="palette-swatch" color="error" />
  <Icon name="palette-swatch" color="success" />
</Playground>

## Icon Library

<Playground>
  <Icon library="Entypo" name="github" />
</Playground>
