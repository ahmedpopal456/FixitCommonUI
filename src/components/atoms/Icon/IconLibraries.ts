import AntDesign from 'react-native-vector-icons/AntDesign';
import Entypo from 'react-native-vector-icons/Entypo';
import EvilIcons from 'react-native-vector-icons/EvilIcons';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import Fontisto from 'react-native-vector-icons/Fontisto';
import Foundation from 'react-native-vector-icons/Foundation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import Zocial from 'react-native-vector-icons/Zocial';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';

export const IconLibraries = {
  AntDesign,
  Entypo,
  EvilIcons,
  Feather,
  FontAwesome,
  FontAwesome5,
  Fontisto,
  Foundation,
  Ionicons,
  MaterialIcons,
  MaterialCommunityIcons,
  Octicons,
  Zocial,
  SimpleLineIcons,
};
