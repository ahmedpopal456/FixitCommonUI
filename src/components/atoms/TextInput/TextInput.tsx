import React from 'react';
import { View, Text } from 'react-native';
import {
  TextInputIconWrapper, TextInputWrapper, IconWrapper, TextBoxWrapper,
} from './style';
import { TextProps } from './TextModel';
import colors from '../../../theme/colors';

export const TextBoxRender:React.FC<TextProps> = (props: TextProps) => {
  if (props.icon != null) {
    return (
      <View>
        <TextBoxWrapper {...props}>
          <IconWrapper style={{ backgroundColor: 'grey' }}>{props.icon}</IconWrapper>
          <TextInputIconWrapper multiline={true}></TextInputIconWrapper>
        </TextBoxWrapper>
      </View>
    );
  }

  return (
    <View>
      <View style={{
        backgroundColor: '#FFF', padding: 5, top: -5, left: 25, zIndex: 50, position: 'absolute',
      }}>
        <Text>{props.title}</Text>
      </View>
      <TextInputWrapper {...props} multiline={true}>
      </TextInputWrapper>
    </View>
  );
};

export const TextBox: React.FC<TextProps> = (props:TextProps) => (
  <TextBoxRender {...props}/>
);

TextBox.defaultProps = {
  borderRadius: 5,
  height: 45,
  borderColor: colors.accent,
  borderWidth: 2,
};
