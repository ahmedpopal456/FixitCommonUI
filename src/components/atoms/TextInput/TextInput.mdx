---
name: TextInput
menu: Atoms
---

import { Props, Playground } from "docz";
import { TextBox, TextBoxIcon } from "./TextInput.tsx";
import { Icon } from '../Icon/Icon.tsx';
import { DoczContainer } from '../../../utils/DoczContainer.tsx'
import { View } from 'react-native';

# TextInput

A Text Input is where users enter relevant information in a form.

# Usage
```tsx
import { TextBox } from "fixit-common-ui";

<TextBox/>
```

## Props
<Props of={TextBox}></Props>

## Example 
<Playground>
  <DoczContainer>
    <TextBox placeholder="Name"/>
    <TextBox title="Section Name" style={{height:80, borderColor: 'grey', borderWidth: 2, outline:'grey'}}/>
    <TextBoxIcon icon={<Icon library="MaterialCommunityIcons" name="currency-usd" color="accent"/>}  style={{height: 45, borderWidth: 2, backgroundColor:'#33343E'}}/>
  </DoczContainer>
</Playground>


