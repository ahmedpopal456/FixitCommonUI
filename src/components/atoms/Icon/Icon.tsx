import React from "react";

import colors from "../../../theme/colors";

import { IconLibraries } from "./IconLibraries";
import { IconProps } from "./icon-props.model";

/** Building the Icon */
export const Icon: React.FC<IconProps> = (props) => {
  const VectorIcon = IconLibraries[props.library!];
  const color: string = colors[props.color!];

  const childProps = { ...props };
  delete childProps.color;

  return <VectorIcon {...childProps} name={props.name} color={color} />;
};

Icon.defaultProps = {
  library: "MaterialCommunityIcons",
  color: "primary",
  size: 24,
};

export default Icon;
