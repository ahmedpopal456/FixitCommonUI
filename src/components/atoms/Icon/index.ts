import { Icon } from './Icon';
import { IconLibraries } from './IconLibraries';

export { Icon, IconLibraries };
