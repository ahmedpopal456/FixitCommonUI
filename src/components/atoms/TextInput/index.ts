import { TextBox } from './TextInput';

export { TextBox };
