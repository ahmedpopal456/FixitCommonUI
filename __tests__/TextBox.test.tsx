import "react-native";
import React from "react";
import { TextBox } from "../src/components/atoms/TextInput";
import { Icon } from "../src/components/atoms/Icon";

import renderer from 'react-test-renderer';

it('matches placeholder snapshot', () => {
    const textInput = renderer.create(<TextBox placeholder="Name"/>);
    expect(textInput.toJSON()).toMatchSnapshot();
})

it('matches default snapshot', () => {
    const textInput = renderer.create(<TextBox/>);
    expect(textInput.toJSON()).toMatchSnapshot();
})

it('matches title snapshot', () => {
    const textInput = renderer.create(<TextBox title="Section Name"/>);
    expect(textInput.toJSON()).toMatchSnapshot();
})

it('matches height snapshot', () => {
    const textInput = renderer.create(<TextBox style={{height:'45'}}/>);
    expect(textInput.toJSON()).toMatchSnapshot();
})

it('matches text input icon snapshot', () => {
    const textInputIcon = renderer.create(<TextBox icon={<Icon library="MaterialCommunityIcons" name="currency-usd" color="accent"/>}/>);
    expect(textInputIcon.toJSON()).toMatchSnapshot();
})


