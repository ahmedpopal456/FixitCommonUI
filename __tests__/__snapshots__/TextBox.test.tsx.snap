// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches default snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "left": 25,
        "padding": 5,
        "position": "absolute",
        "top": -5,
        "zIndex": 50,
      }
    }
  >
    <Text />
  </View>
  <TextInput
    allowFontScaling={true}
    borderColor="#FFD14A"
    borderRadius={5}
    borderWidth={2}
    height={45}
    multiline={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderColor": "#FFD14A",
          "borderRadius": 5,
          "borderWidth": 2,
          "color": undefined,
          "height": 45,
          "justifyContent": "flex-end",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "undefinedpx",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`matches height snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "left": 25,
        "padding": 5,
        "position": "absolute",
        "top": -5,
        "zIndex": 50,
      }
    }
  >
    <Text />
  </View>
  <TextInput
    allowFontScaling={true}
    borderColor="#FFD14A"
    borderRadius={5}
    borderWidth={2}
    height={45}
    multiline={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderColor": "#FFD14A",
          "borderRadius": 5,
          "borderWidth": 2,
          "color": undefined,
          "height": 45,
          "justifyContent": "flex-end",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "undefinedpx",
        },
        Object {
          "height": "45",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`matches placeholder snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "left": 25,
        "padding": 5,
        "position": "absolute",
        "top": -5,
        "zIndex": 50,
      }
    }
  >
    <Text />
  </View>
  <TextInput
    allowFontScaling={true}
    borderColor="#FFD14A"
    borderRadius={5}
    borderWidth={2}
    height={45}
    multiline={true}
    placeholder="Name"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderColor": "#FFD14A",
          "borderRadius": 5,
          "borderWidth": 2,
          "color": undefined,
          "height": 45,
          "justifyContent": "flex-end",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "undefinedpx",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`matches text input icon snapshot 1`] = `
<View>
  <View
    borderColor="#FFD14A"
    borderRadius={5}
    borderWidth={2}
    height={45}
    icon={
      <Icon
        color="accent"
        library="MaterialCommunityIcons"
        name="currency-usd"
        size={24}
      />
    }
    style={
      Array [
        Object {
          "backgroundColor": "#33343E",
          "borderColor": "#FFD14A",
          "borderRadius": 5,
          "borderWidth": 2,
          "color": undefined,
          "flexDirection": "row",
          "height": 45,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "width": "undefinedpx",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 30,
            "justifyContent": "center",
            "marginLeft": 5,
            "marginTop": 5,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "width": 30,
          },
          Object {
            "backgroundColor": "grey",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        library="MaterialCommunityIcons"
        style={
          Array [
            Object {
              "color": "#FFD14A",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Û∞áÅ
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "color": "white",
            "justifyContent": "center",
            "paddingLeft": 10,
            "paddingTop": 10,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`matches title snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "left": 25,
        "padding": 5,
        "position": "absolute",
        "top": -5,
        "zIndex": 50,
      }
    }
  >
    <Text>
      Section Name
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    borderColor="#FFD14A"
    borderRadius={5}
    borderWidth={2}
    height={45}
    multiline={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderColor": "#FFD14A",
          "borderRadius": 5,
          "borderWidth": 2,
          "color": undefined,
          "height": 45,
          "justifyContent": "flex-end",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "undefinedpx",
        },
      ]
    }
    title="Section Name"
    underlineColorAndroid="transparent"
  />
</View>
`;
